viewer/cache.cpp

```diff
@@ -0,0 +1,74 @@
+#include "cache.hpp"
+
+#include <hawktracer/base_types.h>
+
+namespace HawkTracer
+{
+namespace viewer
+{
+
+Cache::Cache()
+{
+}
+
+bool Cache::range_in_cache(HT_TimestampNs start_ts,
+                           HT_TimestampNs stop_ts,
+                           HT_EventKlassId klass_id)
+{
+    if (_events.find(klass_id) == _events.end())
```

Comment:
You do 3 lookups here. If you keep the iterator returned from find() here you only need to do 1